Implementation of the folder navigation by URL

Oleh-Stefanovych opened the pull request:
Now the current folder path is displayed in the URL and handles its change.

The `ESLint` result:
![image](https://user-images.githubusercontent.com/50869178/64737296-2a1b0280-d4f5-11e9-9420-cf8d8686072f.png)

Oleh-Stefanovych commented:
@dkalinin, PTAL.

Oleh-Stefanovych commented on app/presenter/files-presenter/index.js:
```diff
@@ -15,21 +15,42 @@ export class FilesPresenter {
+    if (folderId) {
+      this.store.dispatch('getFolderById', folderId)
+          .then((folder) => {
+            return this._filesPageComponent.loadFolder(folder);
```
- FilePresenter is responsible for loading the folder into which files should be displayed.
- FilePage is responsible for downloading only files.

Accordingly, FilePresenter tells FilePage which folder to load.

Oleh-Stefanovych commented:
@dkalinin, fixed, PTAL.

Fixes:
- Replaced state setting with mutator.
- Removed unnecessary default parameter value.

`Is this PR finalized?` - Yes. Event handling for navigation between components was implemented [here](https://github.com/Oleh-Stefanovych/FileStorage-UI/pull/18).

dkalinin commented on app/store/actions.js:
```diff
@@ -138,4 +141,35 @@ export default {
     };
   },
 
+  getFolderById(store) {
```
Let's rename.

dkalinin commented on app/store/actions.js:
```diff
@@ -138,4 +141,35 @@ export default {
+    };
+  },
+
+  defineFiles(store) {
```
Should be done in init state.

dkalinin commented on app/store/actions.js:
```diff
@@ -138,4 +141,35 @@ export default {
+    };
+  },
+
+  setFileListControlsEnabling(store) {
```
Please remove.

dkalinin commented on app/store/actions.js:
```diff
@@ -138,4 +141,35 @@ export default {
+    };
+  },
+
+  showServerMessage(store) {
```
This action is likely replaced by commit. Server messages are handled in actions.

Oleh-Stefanovych commented:
@dkalinin, fixed, PTAL.

Oleh-Stefanovych commented:
@dkalinin, fixed, PTAL.

Fixes:
- Redirect if URL is "/"
- Set current folder if URL is '/:id'
- Fetch files using displayed folder from state.
- Create separate fields for the different server errors.
- Refactor `Router` class.

dkalinin commented on app/component/high-order-component/files-page/index.js:
```diff
@@ -34,24 +34,71 @@ export class FilesPage extends HighOrderComponent {
+    });
+
+    this.store.eventBus.subscribe('stateChanged.serverError.moveFile', (serverMessage) => {
+      alert(serverMessage);
```
How about to replace alerts with nice error messages?
https://getbootstrap.com/docs/3.4/components/#alerts

Oleh-Stefanovych commented:
@dkalinin, PTAL:

- Implementation of the folder navigation by click was implemented in #18.
- Implementation of the server message display in the style of [bootstrap ](https://getbootstrap.com/docs/3.4/components/#alerts) was implemented in #27.

dkalinin commented on app/router/index.js:
```diff
@@ -25,6 +34,24 @@ export class Router {
+   */
+  get staticRoutingConfigEntries() {
+    const staticRoutingConfig = [];
+    this._routingConfig.forEach((routingEntry) => {
```
Please use JS Array .map()

Oleh-Stefanovych commented:
@dkalinin, PTAL.

Fixes:
- Removed redundant catch blocks.
- Removed redundant error throwing
- Replaced `forEach` block witch `Array.map()`.

**Note**:
- Implementation of the folder navigation by click was implemented in #18. 
- Implementation of the server message display in the style of bootstrap was implemented in #27.
- Implementation of the server error handling was imlemented in #29.

dkalinin commented on app/router/index.js:
```diff
@@ -41,15 +41,12 @@ export class Router {
+      const newRoutingEntry = {};
+      newRoutingEntry.path = routingEntry.path.split(':')[0];
+      newRoutingEntry.presenter = routingEntry.presenter;
+      return newRoutingEntry;
```
ES6 trick:
```js
return this._routingConfig.map(({path, presenter}) => ({
  path: path.split(':')[0],
  presenter
})
```
Not tested!
